.github/workflows/uml.yml

```diff
@@ -0,0 +1,49 @@
+name: Update the UML Diagrams
+
+on:
+  pull_request:
+    branches: [main]
+    # paths:
+    #   - "causalpy/**"
+  push:
+    branches: [main]
+    # paths:
+    #   - "causalpy/**"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+
+      - name: Configure Git Identity
```

Comment:
I am very much a beginner when it comes to GitHub actions, but I've not seen anything about coffin of git identity in the other workflows. Could this be related to the permissions issue?

Or is it required for the later pushed changes?

Reply:
Think it might be if repo doesn't allow for bots to push code. I am seeing that the pre-commit bot didn't push either though the command was recognized